src/doc/examples/range/range.ts

```diff
@@ -0,0 +1,53 @@
+import 'range.css';
+import { template, API, Controller } from '../../../iv';
+import { IvEventEmitter } from '../../../iv/events';
+import { value } from "../../../iv/inputs";
+
+// @@extract: data-definition
+@API class Range {
+    placeholder: number;
+    min:number;
+    max:number;
+    selectEmitter: IvEventEmitter;
+}
+
+// @@extract: controller
+@Controller class RangeCtl {
+    $api: Range;
+    value: number;
+
+    async $init() {
+        this.value = this.$api.placeholder;
+    }
+
+    select() {
+        this.$api.selectEmitter.emit(this.value);
+    }
+}
+
+// @@extract: template
+export const range = template(`($:RangeCtl) => {
+    <div class="range">
+        <input type="range"
+            min={$.min}
+            max={$.max}
+            @value={=$.value}
+            @onmouseup={e=>$.select()}
+            @ontouchend={e=>$.select()}/>
+        <div class="current" style="color: gray">
+            #Current value: {$.value}#
+        </div>
+    </>
+}`, RangeCtl);
+
+const main = template(`(message="", $) => {
+    const onSelect = (selection: number) => {
+        $.message="Selected value: " + selection;
+    };
+    <*range placeholder=30 min=0 max=100
+        @onselect={e=>onSelect(e.data)}
```

Comment:
I would rather have all the event handler inline (and remove the local onSelect function) - to keep things as simple as possible:
@onselect={e=>$.message="Selected value: " + e.data}